fastly-edge-assignments/src/handlers/assignments.rs

```diff
@@ -0,0 +1,175 @@
+        Ok(item) => item,
+        Err(e) => {
+            let (status, message) = match e {
+                KVStoreError::ItemNotFound => {
+                    eprintln!("Missing configuration for SDK key: {}", token_hash);
+
+                    // Return unauthorized if the key does not exist.
+                    // Our protocol lets the client know that the SDK key has not had a UFC
+                    // configuration pre-computed for it in the KV Store.
+                    (StatusCode::UNAUTHORIZED, "Invalid SDK key.".to_string())
+                }
+                _ => {
+                    eprintln!("KV Store error: {:?}", e);
+                    (
+                        StatusCode::INTERNAL_SERVER_ERROR,
+                        "Unexpected KV Store error".to_string(),
+                    )
+                }
+            };
+
+            return Ok(Response::from_status(status).with_body_text_plain(&message));
+        }
+    };
+
+    // Parse the response from the KV store
+    let kv_store_item_body = kv_store_item.take_body();
+    let ufc_config = match UniversalFlagConfig::from_json(
+        SdkMetadata {
+            name: SDK_NAME,
+            version: SDK_VERSION,
+        },
+        kv_store_item_body.into_bytes(),
+    ) {
+        Ok(config) => config,
+        Err(e) => {
+            eprintln!("Failed to parse UFC config: {:?}", e);
+            return Ok(Response::from_status(StatusCode::INTERNAL_SERVER_ERROR)
+                .with_body_text_plain("Invalid configuration format in KV store"));
+        }
+    };
+
+    let configuration = Configuration::from_server_response(ufc_config, None);
+    let configuration = Arc::new(configuration);
+    let flag_keys = configuration.flag_keys();
+    let configuration_store = ConfigurationStore::new();
+    configuration_store.set_configuration(configuration.clone());
+    let evaluator = Evaluator::new(EvaluatorConfig {
+        configuration_store: Arc::new(configuration_store),
+        sdk_metadata: SdkMetadata {
+            name: SDK_NAME,
+            version: SDK_VERSION,
+        },
+    });
+
+    let subject_assignments = flag_keys
+        .iter()
+        .filter_map(|key| {
+            match evaluator.get_assignment(key, &subject_key, &subject_attributes, None) {
+                Ok(Some(assignment)) => FlagAssignment::try_from_assignment(assignment)
+                    .map(|flag_assignment| (key.clone(), flag_assignment)),
+                Ok(None) => None,
+                Err(e) => {
+                    eprintln!("Failed to evaluate assignment for key {}: {:?}", key, e);
+                    None
+                }
+            }
+        })
+        .collect::<HashMap<_, _>>();
+
+    // Create the response
+    let assignments_response = PrecomputedAssignmentsServiceResponse::from_configuration(
+        configuration,
+        subject_assignments,
+    );
```

Comment:
major: computing subject assignments should be part of core. super-weird to have request/response formats in the core but the way to calculate assignment not in the core